src/integrationtests/java/com/aws/greengrass/integrationtests/deviceauth/EvaluateClientDeviceActionsWithPolicyVariablesTest.java

```diff
@@ -0,0 +1,242 @@
+                            .withClientDeviceAuthToken(deviceToken);
+
+            authzClientDeviceAction(ipcClient, connectRequest, true);
+            authzClientDeviceAction(ipcClient, publishRequest, true);
+        }
+    }
+
+    @Test
+    void
+    GIVEN_permissiveGroupPolicyWithThingNameVariable_WHEN_ClientAuthorizesWithThingNameInvalidResource_THEN_ClientNotAuthorized() throws Exception {
+
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerWithAuthorizeClientDeviceActionPermission")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            String deviceToken = getClientDeviceSessionAuthToken("myThing", clientPem);
+
+            AuthorizeClientDeviceActionRequest connectRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:connect").withResource("mqtt:MyCoolThing:foo")
+                            .withClientDeviceAuthToken(deviceToken);
+
+            AuthorizeClientDeviceActionRequest publishRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:publish").withResource("mqtt:topic:SomeThing")
+                            .withClientDeviceAuthToken(deviceToken);
+
+            authzClientDeviceAction(ipcClient, connectRequest, false);
+            authzClientDeviceAction(ipcClient, publishRequest, false);
+        }
+    }
+
+    @Test
+    void GIVEN_permissiveGroupPolicyWithThingNameVariable_WHEN_ClientAuthorizesWithThingNameResourceInvalidAction_THEN_ClientNotAuthorized() throws Exception {
+
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
+                "BrokerWithAuthorizeClientDeviceActionPermission")) {
+            GreengrassCoreIPCClient ipcClient = new GreengrassCoreIPCClient(connection);
+
+            String deviceToken = getClientDeviceSessionAuthToken("myThing", clientPem);
+
+            AuthorizeClientDeviceActionRequest connectRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:connect").withResource("mqtt:topic:myThing")
+                            .withClientDeviceAuthToken(deviceToken);
+
+            AuthorizeClientDeviceActionRequest publishRequest =
+                    new AuthorizeClientDeviceActionRequest().withOperation("mqtt:publish").withResource("mqtt:myThing:foo")
+                            .withClientDeviceAuthToken(deviceToken);
+
+            authzClientDeviceAction(ipcClient, connectRequest, false);
+            authzClientDeviceAction(ipcClient, publishRequest, false);
+        }
+    }
+
+    @Test
+    void GIVEN_permissiveGroupPolicyWithThingNameVariable_WHEN_ClientAuthorizesWithInvalidThingNameResource_THEN_ClientNotAuthorized() throws Exception {
+
+        try (EventStreamRPCConnection connection = IPCTestUtils.getEventStreamRpcConnection(kernel,
```

Comment:
why are we using IPC here? Can't we just call the internal API?

Reply:
true, originally suggested IPC since that's the typical customer entrypoint, but yeah not strictly required